src/transactions/transactions.util.ts

```diff
@@ -140,32 +145,37 @@ export async function processTransaction<
       details,
     };
   } catch (err) {
-    handleSdkError(err);
+    throw handleSdkError(err);
   }
 }
 
-export function handleSdkError(err: unknown): never {
+export function handleSdkError(err: unknown): AppError {
+  if (isAppError(err)) {
```

Comment:
This is important for when errors get re thrown. It didn't solve the unhandled rejection promise I was looking for, but it did fix the "pause" already paused asset.

Although not 100% necessary for this to return, I think its much more flexible to give the choice to the caller compared to throw